src/config-utils.ts

```diff
@@ -5,112 +5,222 @@ import * as yaml from 'js-yaml';
+      throw new Error('Some queries declare multiple languages, their qlpack.yml file is missing or is invalid');
+    }
   }
 }
 
+/**
+ * Get the set of queries included by default.
+ */
+async function addDefaultQueries(languages: string[], resultMap: { [language: string]: string[] }) {
+  const suites = languages.map(l => l + '-code-scanning.qls');
+  await runResolveQueries(resultMap, suites, undefined, false);
+}
+
 // The set of acceptable values for built-in suites from the codeql bundle
 const builtinSuites = ['security-extended', 'security-and-quality'] as const;
-// Derive the union type from the array values
-type BuiltInSuite = typeof builtinSuites[number];
-
-export class Config {
-  public name = "";
-  public disableDefaultQueries = false;
-  public additionalQueries: string[] = [];
-  public externalQueries: ExternalQuery[] = [];
-  public additionalSuites: BuiltInSuite[] = [];
-  public pathsIgnore: string[] = [];
-  public paths: string[] = [];
-
-  public addQuery(configFile: string, queryUses: string) {
-    // The logic for parsing the string is based on what actions does for
-    // parsing the 'uses' actions in the workflow file
-    queryUses = queryUses.trim();
-    if (queryUses === "") {
-      throw new Error(getQueryUsesInvalid(configFile));
-    }
 
-    // Check for the local path case before we start trying to parse the repository name
-    if (queryUses.startsWith("./")) {
-      const localQueryPath = queryUses.slice(2);
-      // Resolve the local path against the workspace so that when this is
-      // passed to codeql it resolves to exactly the path we expect it to resolve to.
-      const workspacePath = fs.realpathSync(util.getRequiredEnvParam('GITHUB_WORKSPACE'));
-      let absoluteQueryPath = path.join(workspacePath, localQueryPath);
-
-      // Check the file exists
-      if (!fs.existsSync(absoluteQueryPath)) {
-        throw new Error(getLocalPathDoesNotExist(configFile, localQueryPath));
-      }
+/**
+ * Parse the suitename to a set of queries and update resultMap.
+ */
+async function parseBuiltinSuite(
+  configFile: string,
+  languages: string[],
+  resultMap: { [language: string]: string[] },
+  suiteName: string) {
+
+  const suite = builtinSuites.find((suite) => suite === suiteName);
+  if (!suite) {
+    throw new Error(getQueryUsesInvalid(configFile, suiteName));
+  }
 
-      // Call this after checking file exists, because it'll fail if file doesn't exist
-      absoluteQueryPath = fs.realpathSync(absoluteQueryPath);
+  const suites = languages.map(l => l + '-' + suiteName + '.qls');
+  await runResolveQueries(resultMap, suites, undefined, false);
+}
 
-      // Check the local path doesn't jump outside the repo using '..' or symlinks
-      if (!(absoluteQueryPath + path.sep).startsWith(workspacePath + path.sep)) {
-        throw new Error(getLocalPathOutsideOfRepository(configFile, localQueryPath));
-      }
+/**
+ * Parse the local path to a set of queries and update resultMap.
```

Comment:
Similarly to the above - you're not parsing the path, you're just using it to find files.  I don't know enough about how the resolve queries function works to be certain about wording here, but how about:

```suggestion
 * Retrieve the set of queries at localQueryPath and add them to resultMap.
```